scripts/dmsdk/gen_csharp.py

```diff
@@ -0,0 +1,300 @@
+# Copyright 2020-2024 The Defold Foundation
```

Comment:
The script that outputs C# generated code.

Reply:
Does it make sense to add a comment into file that macros/defines do not translate well for now? 
I think we want to improve it in the future?

Reply:
Will it ever work though?
How to automatically reimplement from C to C# in these cases?
I think it's just too tricky. I think having a few manually curated reimplementations is the way to go here.